adding nvs_commit to set_blob for P11_Cert (CA-359)

matthewsdan opened the pull request:
P11_Cert and P11_Key were not being committed during set_blob

matthewsdan committed:
adding nvs_commit to set_blob for P11_Cert

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/espressif/esp-aws-iot?pullRequest=243) <br/>All committers have signed the CLA.

matthewsdan committed:
remove partition erase from read error

sometimes a read error is not fatal and can be recovered in reset
however and erased partition can never recover!

matthewsdan reviewed (commented):
This one is higher risk than the nvs_commit change in the same PR. The erase fails to consider a bad handle and might cause the partition to be erased when a new handle or reboot might have fixed it. Erasing the partition is drastic.
